conn2res/plotting.py

```diff
@@ -591,11 +599,11 @@ def plot_performance(
     ax = fig.subplots(1, 1)
 
     # set color palette
+    col_defaults = {'palette': 'husl'}
     if hue is not None:
-        n_hues = len(np.unique(df[hue]))
-        palette = sns.color_palette('husl', n_hues+1)[:n_hues]
-    else:
-        palette = sns.color_palette('husl')
+        col_defaults.update(n_colors=len(pd.unique(df[hue])))
+    col_defaults.update(**col_params)
+    palette = sns.color_palette(**col_defaults)
```

Comment:
This changes the behaviour of the code slightly as it uses the full colour palette by default. However, it is easy to get the same functionality as before by calling plot_performance with col_params argument and passing n_colours with a higher value than the number of hues. For instance, if you have 6 number of hues and want to use `husl` palette then you would set `col_params={'palette': 'husl', 'n_colors': 7}`.